# Traversability map parameters
submap_service: "/elevation_mapping_long_range/get_submap"
elevation_map/topic: "/image_to_gridmap_demo/grid_map"
robot_frame_id: "base"
min_update_rate: 0.1
map_center_x: 1.0
map_center_y: 0.0
map_length_x: 6.0
map_length_y: 6.0
circular_footprint_offset: 0.15

traversability_map:
  footprint_polygon: [[0.32, 0.29], [0.32, -0.29], [-0.32, -0.29], [-0.32, 0.29]]
  frame_id: map
  traversability_default: 0.1

# Traversability map filter parameters
traversability_map_filters:
 - name: surfaceNormalsFilter
   type: traversabilityFilters/SurfaceNormalsFilter
   params:
    estimation_radius: 0.1
    surface_normal_positive_axis: "z"
 - name: slopeFilter
   type: traversabilityFilters/SlopeFilter
   params:
    map_type: traversability_slope
    critical_value: 0.610865
 - name: stepFilter
   type: traversabilityFilters/StepFilter
   params:
    map_type: traversability_step
    critical_value: 0.12
    first_window_radius: 0.08
    second_window_radius: 0.08
    critical_cell_number: 5
 - name: roughnessFilter
   type: traversabilityFilters/RoughnessFilter
   params:
    map_type: traversability_roughness
    critical_value: 0.02
    estimation_radius: 0.05
 - name: weightedSumFilter
   type: gridMapFilters/WeightedSumFilter
   params:
    layers: [traversability_slope, traversability_step, traversability_roughness]
    weights: [2, 2, 1]
    layer_out: traversability
    normalize: 1
# - name: minFilter
#   type: gridMapFilters/MinFilter
#   params:
#    layers: [traversability_slope, traversability_step, traversability_roughness]
#    layer_out: traversability
# - name: lowerThresholdFilter
#   type: gridMapFilters/ThresholdFilter
#   params:
#    lower_threshold: 0.2
#    set_to: 0.0
#    layers: [traversability]
# - name: upperThresholdFilter
#   type: gridMapFilters/ThresholdFilter
#   params:
#    upper_threshold: 0.9
#    set_to: 1.0
#    layers: [traversability]
 - name: deletionFilter
   type: gridMapFilters/DeletionFilter
   params:
    layers: [variance, surface_normal_x, surface_normal_y, surface_normal_z]